nautobot/docs/development/core/ui-component-framework.md

```diff
@@ -0,0 +1,408 @@
+
+```python
+from nautobot.apps.ui import (
+    NautobotUIViewSet,
+    ObjectDetailContent,
+    ObjectFieldsPanel,
+    SectionChoices
+)
+
+class ExampleUIViewSet(NautobotUIViewSet):
+    queryset = Example.objects.all()
+    ...
+    
+    object_detail_content = ObjectDetailContent(
+        panels=[
+            ObjectFieldsPanel(
+                section=SectionChoices.LEFT_HALF,
+                weight=100,
+                fields=["name", "status", "type"]
+            )
+        ]
+    )
+```
+
+//TODO: (include image) <img src="/api/placeholder/600/300" alt="Basic Panel Layout" />
+*Basic panel layout with ObjectFieldsPanel*
+
+## Core Concepts
+
+### ViewSet Configuration
+The UI Framework is built around the concept of configurable ViewSets. Each ViewSet defines:
+- Query handling
+- Panel layouts
+- Content organization
+- Tab structure
+
+### Panels
+Panels are the building blocks of your UI. They contain specific types of content and can be positioned within sections.
+
+//TODO: (include image) <img src="/api/placeholder/700/350" alt="Panel Structure" />
+*Panel structure showing different sections and layouts*
+
+## Panel Types
+
+### Base Panel Properties
+All panel types inherit these base properties:
+
+```python
+Panel(
+    # Visual Organization
+    label="",                    # Text label shown in panel header
+    section=SectionChoices.FULL_WIDTH,  # Panel placement in layout
+    weight=100,                  # Order within section (lower numbers appear first)
+    
+    # HTML/Template Configuration
+    body_id=None,                # HTML ID for the panel body element
+    body_content_template_path=None,  # Template for panel content
+    header_extra_content_template_path=None,  # Additional header content
+    footer_content_template_path=None,  # Footer content template
+    template_path="components/panel/panel.html",  # Overall panel template
+    body_wrapper_template_path="components/panel/body_wrapper_generic.html"  # Body wrapper template
+)
+```
+
+### 1. KeyValueTablePanel
+Displays custom key-value pairs with optional transformations.
+
+```python
+KeyValueTablePanel(
+    # Data Configuration
+    data={                      # Dictionary of key-value pairs to display
+        "speed": "1000000",
+        "notes": "**Important**"
+    },
+    
+    hide_if_unset=(),          # Keys to hide if value is falsey
+                               # Example: ["optional_field", "notes"]
+    
+    value_transforms={          # Transform functions for specific keys
+        "speed": [             # List of functions to apply in order
+            humanize_speed,    # Convert 1000000 to "1 Gbps"
+            placeholder        # Show placeholder if empty
+        ],
+        "notes": [
+            render_markdown,   # Convert markdown to HTML
+            placeholder       # Show placeholder if empty
+        ]
+    },       # Order within section
+)
+```
```

Comment:
We might need two "examples" for each panel class - one just showing the parameters and their default values, and another showing an example of real usage. I think mixing the two in a single example is confusing.